src/Cart/Cart.js

```diff
@@ -0,0 +1,149 @@
+import {
+  deleteItemFromCartApi,
+  incrementQuantityFromCartApi,
+  decrementQuantityFromCartApi,
+} from "../api-calls";
+
+import "./Cart.css";
+import { Link } from "react-router-dom";
+import { findUserById } from "../utils";
+import { useAuth } from "../Auth";
+
+export const Cart = () => {
+  const { userState, userDispatch } = useUser();
+  const { user } = useAuth();
+  const currentUser = findUserById(userState, user._id);
+
+  function totalPrice() {
+    return currentUser?.cart.reduce((acc, value) => {
+      return acc + value.quantity * value.price;
+    }, 0);
+  }
+
+  const totalItemPrice = (item) => {
+    return item.quantity * item.price;
+  };
+
+  return (
+    <div className='products products-cart'>
+      {currentUser?.cart.length === 0 ? (
+        <div className='card cart-empty-card'>
+          <h3>Your Cart is Empty</h3>
+          <hr className='hr' />
+          <p className='mg-1'>There are no items in your cart.</p>
+          <Link to='/products'>
+            <button className='btn primary btn-shop'>Shop Now</button>
+          </Link>
+        </div>
+      ) : (
+        <div className='products products-cart'>
+          <div>
+            {currentUser?.cart.map((product) => {
+              return (
+                <div className='card-horizontal' key={product._id}>
+                  <img
+                    className='card-horizontal-img'
+                    src={product.image}
+                    alt=''
+                  />
+                  <button
+                    onClick={() =>
+                      deleteItemFromCartApi(currentUser, product, userDispatch)
+                    }
+                    className='floating-act badge-close tertiary'
+                  >
+                    <i className='fas fa-lg fa-times'></i>
+                  </button>
+                  <div className='card-info'>
+                    <h3>{product.name}</h3>
+                    <small>{product.color}</small>
+                    <div className='card-details-horizontal'>
+                      <div className='card-quant'>
+                        <button
+                          onClick={() =>
+                            incrementQuantityFromCartApi(
+                              currentUser,
+                              product,
+                              userDispatch
+                            )
+                          }
+                          className='floating-act badge-close tertiary'
+                        >
+                          <i className='fas fa-lg fa-plus'></i>
+                        </button>
+                        <p>{product.quantity}</p>
+                        <button
+                          onClick={() =>
+                            decrementQuantityFromCartApi(
+                              currentUser,
+                              product,
+                              userDispatch
+                            )
+                          }
```

Comment:
when I click this button on UI , the snackbar is showing me "incrementing the item " instead of "decrementing"

Reply:
Done